### PR TITLE
feature: add outbound_proxy_host/port settings, to support HTTP proxy via curl

### PR DESCRIPTION
**Pull request by bstewart00**
### Purpose/goal of PR (Issue #)
https://wovnio.atlassian.net/browse/ST-2444
### Comments
Support connecting to wovn servers via HTTP proxy. Only supported for curl.

Didn't test with an actual proxy server but curl reported this error so the setting is being applied correctly
```
curl_error($curl_session)
"Failed to connect to proxy.com port 7979: Connection timed out"
```

### CODE REVIEW

**Review by iwai-wovn — APPROVED** [review]

File: src/wovnio/wovnphp/Store.php [hunk 1]
```diff
@@ -260,4 +264,19 @@ public function compressApiRequests()
...
+
+        if (isset($this->settings['outbound_proxy_host'])) {
+            $result = $this->settings['outbound_proxy_host'];
+        }
+        if (isset($this->settings['outbound_proxy_port'])) {
+            $port = $this->settings['outbound_proxy_port'];
+            $result .= ":$port";
+        }
+
+        return $result;
```
Comment (Soichiro725) [inline src/wovnio/wovnphp/Store.php:280]: Why do we need to split host and port? How about having configuration to be connected.
e.g. `outbound_proxy_host_and_port: site.com:8000`

Comment (bryanzerrudo) [reply src/wovnio/wovnphp/Store.php:280]: Seems like there's a `CURLOPT_PROXYPORT`:
https://www.php.net/manual/en/function.curl-setopt.php (under "options")

If we want, we can use that to specify the port.

Comment (bstewart00) [reply src/wovnio/wovnphp/Store.php:280]: to be consistent with wovnjava 😎

Comment (bstewart00) [reply src/wovnio/wovnphp/Store.php]: They do mention this re CURLOPT_PROXYPORT
> The proxy's port number may optionally (but discouraged) be specified with the separate option [CURLOPT_PROXYPORT](https://curl.se/libcurl/c/CURLOPT_PROXYPORT.html).

**Review by Soichiro725 — APPROVED** [review]

**Pull request merged**